libcontainer/cgroups/systemd/unified_hierarchy.go

```diff
@@ -167,18 +167,39 @@ func (m *UnifiedManager) Destroy() error {
 	if m.Cgroups.Paths != nil {
 		return nil
 	}
+	dirPath, err := m.GetUnifiedPath()
+	if err != nil {
+		return err
+	}
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
 	dbusConnection, err := getDbusConnection()
 	if err != nil {
 		return err
 	}
-	dbusConnection.StopUnit(getUnitName(m.Cgroups), "replace", nil)
-	if err := cgroups.RemovePaths(m.Paths); err != nil {
-		return err
+	unitName := getUnitName(m.Cgroups)
+	statusChan := make(chan string, 1)
+	if _, err := dbusConnection.StopUnit(unitName, "replace", statusChan); err == nil {
```

Comment:
@kolyshkin 
I think we may not need `StopUnit` method. Because if the init process has been exited, the cgroup path will be deleted by systemd automatically.
WDYT

Reply:
Well, I think it is still needed, otherwise it will show up in systemctl output as "killed" or something like that (I haven't checked yet, just a hunch).

We can also take a look at crun implementation -- in there the unit is removed from systemd first, then killed. Might make more sense.